### ISSUE TITLE
make tag use cases more obvious

### ISSUE THREAD

**Issue by lorenzo-cavazzi** [SwissDataScienceCenter/renku-python#1264]
**Is your improvement request related to a use case or a component design? Please describe.**

The UX around dataset improved a lot lately, that's amazing!
One nice feature is the possibility to tag datasets, but I don't understand well how they should work. Should they work as:
* "git like" tags? Not sure it would work well, since there is no easy whay to checkout a tagged version (I can't "checkout" a specific tag for the datasets, right?)
* markers? It makes sense, but then I would like to see their name listed on `renku dataset`, leaving `ls-tags` for the details.
* versions? The example in the docs mention a version number `renku dataset tag my-dataset 1.0 -d "Version 1.0 tag"` but I also see a `VERSION` column in `renku dataset`. This contributes further in making the scope of tags unclear.

**Comment by Panaetius** [comment]
> "git like" tags? Not sure it would work well, since there is no easy whay to checkout a tagged version (I can't "checkout" a specific tag for the datasets, right?)

Yes something along those lines, or Docker tags. They can be used for versioning, but it's really up to the user how they want to use them. The tags themselves do include commit information, so we could have a command that uses a dataset at a specific tag. The only command that currently supports this is `dataset export` that allows exporting a specific tag instead of the current version.

The original motivation was that we wanted a feature to version datasets but that there's no single versioning scheme we would want to impose on users, rather that they each can pick whichever scheme suits them.

With the new dataset immutability PR where each change to a dataset generates a new version, tags could become more relevant. @mohammad-sdsc 

In this sense, a dataset tag is a way to create a named pointer to a specific version of a dataset, also allowing to add some semantics to a specific point in time of a dataset.

 Maybe we should have a design discussion on what commands we'd like to have that work with tags, especially in view of the dataset immutability that hopefully will be merged soon
